packages/core/src/code-gen/parsers/generateReferenceParser.ts

```diff
@@ -7,104 +7,95 @@ import {
 import { ParserModelType, ReferenceParserModel } from "./generateParserModel"
 import { generateParserBody } from "./generateParser"
 import generateParserFromModel from "./generateParserFromModel"
-import { capitalize } from "../../utils"
+import * as tsx from "../../tsx"
 
 export default function generateReferenceParser(
   pointer: Pointer<ReferenceParserModel>,
 ): ts.Statement {
-  const hasNoTypeArgs = pointer.model.typeArguments.length === 0
-  return ts.factory.createIfStatement(
-    hasNoTypeArgs
-      ? generateReferenceValidator(pointer)
-      : generateReferenceValidatorWithTypeArguments(pointer),
-    // TODO populate the errors with the actual errors
-    generatePushErrorExpressionStatement(
-      pointer.errorPath,
-      `not a ${pointer.model.typeName}`,
-    ),
-    assignDataToResult(pointer.resultVarExpr, pointer.dataVarExpr),
+  const parseResultName = `parseResult`
+  return tsx.block(
+    tsx.const({
+      name: parseResultName,
+      init: generateParserCall(pointer),
+    }),
+    tsx.statement.if({
+      expression: tsx.expression.negate(
+        tsx.expression.propertyAccess(parseResultName, "ok"),
+      ),
+      // TODO populate the errors with the actual errors
```

Comment:
Is this something we need to fix now, or write down to not forget?

Reply:
The TODO was already there. I intent to fix this later.